src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs

```diff
@@ -109,8 +122,42 @@ public Transaction ToTransaction()
             tx.SenderAddress = From;
             tx.Value = Value ?? 0;
             tx.Data = Data ?? Input;
+            tx.Type = (TxType)Type;
+            tx.AccessList = TryGetAccessList();
 
             return tx;
         }
+
+        private AccessList? TryGetAccessList()
+        {
+            if (Type == 1 && AccessList != null)
+            {
+                return GetAccessList();
+            }
+            return null;
+        }
+
+        private AccessList GetAccessList()
+        {
+            AccessListBuilder accessListBuilder = new();
+            foreach (var item in AccessList)
+            {
+                accessListBuilder.AddAddress(item.Address);
+                foreach (var storageKey in item.StorageKeys)
+                {
+                    accessListBuilder.AddStorage(storageKey);
+                }
```

Comment:
for will be faster

Reply:
can we reuse AccessListBuilder code here?